src/core/Akka.Remote/Serialization/MessageContainerSerializer.cs

```diff
@@ -126,20 +129,21 @@ private byte[] SerializeSelection(ActorSelectionMessage sel)
         /// </exception>
         public override object FromBinary(byte[] bytes, Type type)
         {
-            SelectionEnvelope selectionEnvelope = SelectionEnvelope.ParseFrom(bytes);
-            var manifest = selectionEnvelope.HasMessageManifest ? selectionEnvelope.MessageManifest.ToStringUtf8() : string.Empty;
+            SelectionEnvelope selectionEnvelope = SelectionEnvelope.Parser.ParseFrom(bytes);
+            var manifest = !selectionEnvelope.MessageManifest.IsEmpty ? selectionEnvelope.MessageManifest.ToStringUtf8() : string.Empty;
+            //TODO the line above must be the same as selectionEnvelope.MessageManifest.ToStringUtf8()
```

Comment:
It was a comment to myself about code simplification. Once we have tests I will process this comment.